plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -8510,6 +8511,117 @@ public void testCreateAcidTableUnsupported()
+                        ")");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testDuplicateExtraProperties()
+    {
+        String tableName = format("%s.%s.test_duplicate_extra_properties", getSession().getCatalog().get(), getSession().getSchema().get());
+        @Language("SQL") String createTableSql = format("""
+                        CREATE TABLE %s (
+                          c1 integer)
+                        WITH (
+                          extra_properties = MAP(ARRAY['extra.property', 'extra.property'], ARRAY['true', 'false']),
+                          format = 'ORC'
+                        )""",
+                tableName);
+        assertQueryFails(createTableSql, "Invalid value for catalog 'hive' table property 'extra_properties': Cannot convert.*");
+    }
+
+    @Test
+    public void testOverwriteExistingPropertyWithExtraProperties()
+    {
+        String tableName = format("%s.%s.test_overwrite_extra_properties", getSession().getCatalog().get(), getSession().getSchema().get());
+        @Language("SQL") String createTableSql = format("""
+                        CREATE TABLE %s (
+                          c1 integer)
+                        WITH (
+                          extra_properties = MAP(ARRAY['transactional'], ARRAY['true']),
+                          format = 'ORC'
+                        )""",
+                tableName);
+        assertThatThrownBy(() -> assertUpdate(createTableSql))
+                .isInstanceOf(QueryFailedException.class)
+                .hasMessage("Invalid keys in extra_properties: transactional");
+    }
+
+    @Test
+    public void testAddingNullProperty()
+    {
+        String tableName = format("%s.%s.test_duplicate_extra_properties", getSession().getCatalog().get(), getSession().getSchema().get());
+        @Language("SQL") String createTableSql = format("""
+                        CREATE TABLE %s (
+                          c1 integer)
+                        WITH (
+                          extra_properties = MAP(ARRAY['null.property'], ARRAY[null]),
+                          format = 'ORC'
+                        )""",
+                tableName);
+        assertQueryFails(createTableSql, ".*Extra table property value cannot be null '\\{null.property=null}'.*");
+    }
+
+    @Test
+    public void testAddingEmptyProperty()
+    {
+        String tableName = format("%s.%s.test_duplicate_extra_properties", getSession().getCatalog().get(), getSession().getSchema().get());
+        @Language("SQL") String createTableSql = format("""
+                        CREATE TABLE %s (
+                          c1 integer)
+                        WITH (
+                          extra_properties = MAP(),
+                          format = 'ORC'
+                        )""",
+                tableName);
+        assertQueryFails(createTableSql, ".*Extra table properties cannot be empty.*");
+    }
+
```

Comment:
Add an extra testcase for the following scenario

```
trino:default> create table test4 (c1 integer) 
            -> with (
            ->      extra_properties = map(array['MiXEd'], array['value1']));

trino:default> select "mixed" from "test4$properties";
 mixed  
--------
 value1 
(1 row)

trino:default> select "MiXEd" from "test4$properties";
 MiXEd  
--------
 value1 
(1 row)
```

We showcase here the fact that the user can retrieve, independently of the case used, the value for a table property

Reply:
I'm not sure if we need to add this test case - as it shows current limitation we have today i.e all names are converted to lower case for the connector.